internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -273,6 +275,14 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 	return deployment, nil
 }
 
+func expectedTerminationGracePeriodSeconds(cfg *egv1a1.ShutdownConfig) *int64 {
```

Comment:
shouldn't this be equal to `DrainTimeout` ?

Reply:
`terminationGracePeriodSeconds` needs to be higher than the drain timeout so the pod has time to handle the SIGTERM before being sent a SIGKILL. It doesn't have to be much, but it does need to be marginally higher, and assuming other timeouts are functioning it should never be reached under normal circumstances.